lib/init.js

```diff
@@ -0,0 +1,141 @@
+'use babel';
+
+// eslint-disable-next-line import/no-extraneous-dependencies, import/extensions
+import { CompositeDisposable } from 'atom';
+
+let path;
+let helpers;
+
+const pattern = '.+:(?<line>\\d+):(?<col>\\d+)-(?<colEnd>\\d+):' +
+  ' \\((?<type>[EW])\\d+\\) (?<message>.*)';
+
+function loadDeps() {
+  if (!helpers) {
+    helpers = require('atom-linter');
+  }
+  if (!path) {
+    path = require('path');
+  }
+}
+
+function checkedOption(params, opt, args) {
+  if (args.length > 0) {
+    params.push(opt);
+    return params.concat(args);
+  }
+
+  return params;
+}
+
+function reportToMessage(report, file) {
+  const end = report.range[1];
```

Comment:
Luacheck does always return a column end (because `--range` is passed in the command line), which is inclusive, that is, the number returned is the position of the last character (All numbers returned by Luacheck start at 1 and not at 0). Maybe Ranges are exclusive so you actually need the position of the first character not included in the Range? I don't know

http://luacheck.readthedocs.io/en/stable/cli.html#stable-interface-for-editor-plugins-and-tools